PokeStar/PokeStar/DataModels/Raid.cs

```diff
@@ -1,131 +1,120 @@
-      private const int maxInvites = 10;
       public string Location { get; set; }
       public string Time { get; set; }
       public short Tier { get; set; }
       public RaidBoss Boss { get; private set; }
-      public int PlayerCount { get; private set; }
-      public int HereCount { get; private set; }
-      public Dictionary<SocketGuildUser, int> Attending { get; private set; }
-      public Dictionary<SocketGuildUser, int> Here { get; private set; }
-      public Dictionary<SocketGuildUser, int> Invite { get; private set; }
+      public List<RaidGroup> Groups { get; private set; }
+      public List<string> RaidBossSelections { get; set; }
       public DateTime CreatedAt { get; private set; }
+      private Dictionary<SocketGuildUser, int> Invite { get; set; }
 
       public Raid(short tier, string time, string location, string boss = null)
       {
          Tier = tier;
          Time = time;
          Location = location;
-         PlayerCount = 0;
-         HereCount = 0;
          SetBoss(boss);
-         Attending = new Dictionary<SocketGuildUser, int>();
-         Here = new Dictionary<SocketGuildUser, int>();
+         Groups = new List<RaidGroup>
+         {
+            new RaidGroup()
+         };
          Invite = new Dictionary<SocketGuildUser, int>();
+         RaidBossSelections = new List<string>();
          CreatedAt = DateTime.Now;
       }
 
-      public void PlayerAdd(SocketGuildUser player, int partySize, bool isInvite = false)
+      public ImmutableDictionary<SocketGuildUser, int> GetReadonlyInvite()
+      {
+         return Invite.ToImmutableDictionary(k => k.Key, v => v.Value);
+      }
+
+      public bool PlayerAdd(SocketGuildUser player, int partySize, SocketGuildUser user = null)
       {
-         if (Attending.ContainsKey(player))
+         int group;
+         if (user == null)
          {
-            int newPlayerCount = PlayerCount + (partySize - Attending[player]);
-            if (newPlayerCount <= playerLimit)
-            {
-               Attending[player] = partySize;
-               PlayerCount = newPlayerCount;
-            }
+            group = IsInRaid(player);
+            if (group == -1)
+               group = FindSmallestGroup();
+            Groups.ElementAt(group).Add(player, partySize);
          }
-         else if (Here.ContainsKey(player))
+         else // is invite
          {
-            int newPlayerCount = PlayerCount + (partySize - Here[player]);
-            if (newPlayerCount <= playerLimit)
+            group = IsInRaid(user);
+            if (group != -1)
             {
-               HereCount += partySize - Here[player];
-               PlayerCount = newPlayerCount;
-               Here[player] = partySize;
+               Groups.ElementAt(group).Invite(player, user);
+               return false;
             }
          }
-         else if (Invite.ContainsKey(player))
+         var newGroup = Groups.ElementAt(group).SplitGroup();
```

Comment:
heck

Reply:
consider putting the checking of player group size back in raidgroup.cs

Reply:
It will not split on every add, Line 120-121 in RaidGroup will return null if the group should not be split. if the value returned is null then a new group will not be added to Groups